gcp/opentelemetry-demo-features.yaml.gotmpl

```diff
@@ -0,0 +1,74 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# this file is used by Helm chart launch steps.
+# when merging YAML with the default values, objects are merged but lists are overwritten.
+# so, where a list is used, we have copied the defaults from the upstream values to preserve them.
+
+opentelemetry-collector:
+  # Service account for workload identity
+  serviceAccount:
+{{ with .Values | getOrNil "workload_identity_project_id" }}
+    annotations:
+      "iam.gke.io/gcp-service-account": "opentelemetry-demo@{{ . }}.iam.gserviceaccount.com"
+{{ end }}
+
+  image:
+{{ with .Values | getOrNil "collector_image_repository" }}
+    repository: "{{ . }}"
+{{ end }}
+{{ with .Values | getOrNil "collector_image_tag" }}
+    tag: "{{ . }}"
+{{ end }}
```

Comment:
@quentinmit can you please show what this section should look like? I can't get it to work with an empty image, keep getting `[ERROR] 'image.repository' must be set`

Reply:
Hm. The default value comes from the [helm chart's values.yaml](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/f903096983044e2d4d8c5cf25d9510c1e469ad0e/charts/opentelemetry-demo/values.yaml#L19-L25):

```yaml
  image:
    repository: ghcr.io/open-telemetry/demo
    # Overrides the image tag whose default is the chart appVersion.
    # The service's name will be applied to the end of this value.
    tag: ""
    pullPolicy: IfNotPresent
    pullSecrets: []
```

I believe anything we specify here should merge with that upstream YAML. I would expect the following to work:

```yaml
{{ if or (.Values | getOrNil "collector_image_repository") (.Values | getOrNil "collector_image_tag") }}
  image:
{{ with .Values | getOrNil "collector_image_repository" }}
    repository: "{{ . }}"
{{ end }}
{{ with .Values | getOrNil "collector_image_tag" }}
    tag: "{{ . }}"
{{ end }}
{{ end }}
```